### ISSUE TITLE
Define box shadows as part of theme instead of constants

### ISSUE THREAD

**Issue by michalkvasnicak** [Byteclaw/visage#389]
Basically now it's impossible to redefine internal box shadows used for pseudo states.

**Comment by michalkvasnicak** [comment]
This one maybe can be solved in #397 but now in a way that we'll extract box shadows to theme but rather add a mixin and face for each pseudostate.

**Comment by jurajhrib** [comment]
I mean, it makes sense to have shadow constants in the theme and then use them in face/mixins.

**Comment by michalkvasnicak** [comment]
@jurajhrib what do you mean? Can you give me an example?

**Comment by jurajhrib** [comment]
Theme {
  surfaceShadow: '...',
  controlShadow: '...',
  ...
}

Then use for example in the button mixin or face 
boxShadow: surfaceShadow

Or am I missing something?

**Comment by michalkvasnicak** [comment]
Yes this is one I was thinking about too. But we need to make them optional and provide defaults so we don't force users to specify them. Could you take a look into this issue please?
